core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -75,6 +75,10 @@
     public static final String SAFE_PARAMETERS_SYSTEM_PROPERTY_NAME = ParametersAction.class.getName() +
             ".safeParameters";
 
+    @Restricted(NoExternalUse.class)
+    public static final String DONT_KEEP_UNDEFINED_PARAMETERS_SYSTEM_PROPERTY_NAME = ParametersAction.class.getName() +
```

Comment:
Some Javadoc is would be useful here (though I cannot request it since properties above do not have it). E.g. brief description of the behavior and `@since TODO`